metronome/checkpointing/models/src/io/iohk/metronome/checkpointing/models/Transaction.scala

```diff
@@ -13,9 +14,13 @@ import scodec.bits.BitVector
   * we support, which is to register proposer blocks in the ledger, required by Advocate,
   * and to register checkpoint candidates.
   */
-sealed trait Transaction
+sealed trait Transaction {
+  def hash: Transaction.Hash
+}
 
 object Transaction {
+  object Hash extends Tagger[ByteVector]
+  type Hash = Hash.Tagged
```

Comment:
This should be enough, and then you don't need the overrides below:
```scala
sealed trait Transaction extends RLPHash[Transaction, Transaction.Hash]

object Transaction
    extends RLPHashCompanion[Transaction]()(RLPCodecs.rlpTransaction) {
```

It will include the transaction type tag in the RLP representation of the transaction instances.